File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float userChange;

    // Only allow positive input
    do
    {
        userChange = get_float("Change owed: ");
    }
    while (userChange < 0);

    // Round up the cents after conversion
    int cents = round(userChange * 100);
    int numberOfCoins = 0;

    // Reduce number of cents, going highest first
    // while counting number of coins requires
    while (cents > 0)
    {
        if (cents >= 25)
        {
            cents -= 25;
        }
        else if (cents >= 10)
        {
            cents -= 10;
        }
        else if (cents >= 5)
        {
            cents -= 5;
        }
        else
        {
            cents -= 1;
        }

        numberOfCoins ++;
    }

    printf("%i\n", numberOfCoins);
}

File: pset6/cash/cash.py
# Refactor of pset1/cash
from cs50 import get_float


def main():
    while True:
        change = get_float("Change owed: ")

        # Keep asking until valid number is provided
        if (change > 0):
            break

    # Print number of coins
    print(coins_num(change))


def coins_num(change):
    coin_count = 0
    cents = round(change * 100)

    # Reduce the change starting with larger coins
    while cents > 0:
        if cents >= 25:
            cents -= 25
        elif cents >= 10:
            cents -= 10
        elif cents >= 5:
            cents -= 5
        else:
            cents -= 1

        coin_count += 1

    return coin_count


if __name__ == "__main__":
    main()